### PR TITLE
feat: Add `EmailTransportAppengine` as default email transport class

### PR DESCRIPTION
**Pull request by phorward**
Brings back the default Google AppEngine Mail API as an abstracted `EmailTransport`-class.

### CODE REVIEW

File: src/viur/core/email.py [hunk 1]
```diff
@@ -508,3 +509,50 @@ def deliverEmail(*, sender: str, dests: list[str], cc: list[str], bcc: list[str]
...
+                Attachment(attachment["filename"], attachment["content"])
+                for attachment in attachments
+            ]
+
+        SendMail(**params)
+
+
+# Set (limited, but free) Google AppEngine Mail API as default
+if conf.email.transport_class is None:
+    conf.email.transport_class = EmailTransportAppengine
```
Comment (sveneberth) [inline src/viur/core/email.py:558]: can't we set this directly in the config?

**Review by sveneberth — CHANGES_REQUESTED** [review]

File: src/viur/core/email.py (hunk 1 above)
Comment (phorward) [reply src/viur/core/email.py:558]: I think this isn't possible due circular imports.

Comment (sveneberth) [reply src/viur/core/email.py:558]: Well, okay.

**Review by sveneberth — COMMENTED** [review]
So far this is okay for me.

But on local develop server it's not working. How should we handle this?

```python
[2024-07-12 18:16:38,742] /.../viur/core/skeleton.py:1552 [ERROR] Failed to notify mail@me.de
Traceback (most recent call last):
  File "/project/viur-core/src/viur/core/skeleton.py", line 1550, in handleFinish
    email.sendEMail(dests=customData["notify"], stringTemplate=txt, skel=None)
  File "/project/viur-core/src/viur/core/email.py", line 282, in sendEMail
    sendEmailDeferred(queueEntity.key, _queue="viur-emails")
  File "/project/viur-core/src/viur/core/tasks.py", line 627, in wrapper
    return make_deferred(func, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/viur-core/src/viur/core/tasks.py", line 543, in make_deferred
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/viur-core/src/viur/core/email.py", line 130, in sendEmailDeferred
    resultData = transportClass.deliverEmail(dests=queueEntity["dests"],
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/viur-core/src/viur/core/email.py", line 553, in deliverEmail
    SendMail(**params)
  File "/venv/lib/python3.11/site-packages/google/appengine/api/mail.py", line 401, in send_mail
    message.send(make_sync_call=make_sync_call)
  File "/venv/lib/python3.11/site-packages/google/appengine/api/mail.py", line 1209, in send
    make_sync_call('mail', self._API_CALL, message, response)
  File "/venv/lib/python3.11/site-packages/google/appengine/api/apiproxy_stub_map.py", line 96, in MakeSyncCall
    return stubmap.MakeSyncCall(service, call, request, response)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/google/appengine/api/apiproxy_stub_map.py", line 353, in MakeSyncCall
    rpc.MakeCall(service, call, request, response)
  File "/venv/lib/python3.11/site-packages/google/appengine/api/apiproxy_rpc.py", line 133, in MakeCall
    self._MakeCallImpl()
  File "/venv/lib/python3.11/site-packages/google/appengine/runtime/default_api_stub.py", line 181, in _MakeCallImpl
    raise apiproxy_errors.RPCFailedError(
google.appengine.runtime.apiproxy_errors.RPCFailedError: Attempted RPC call without active security ticket

```

**Comment by phorward** [comment]
> But on local develop server it's not working. How should we handle this?

Yes, you're right. In this case, I would always log a message, at is the case with `conf.email.send_from_local_development_server = False`. I can integrate this into this PR when #1212 is merged.

**Comment by phorward** [comment]
> I can integrate this into this PR when #1212 is merged.

Added by [634ad83](https://github.com/viur-framework/viur-core/pull/1210/commits/634ad837a109b30555bc696286f8ab679ae725d2)

File: src/viur/core/email.py [hunk 2]
```diff
@@ -584,3 +587,53 @@ def deliverEmail(*, sender: str, dests: list[str], cc: list[str], bcc: list[str]
...
+    def deliverEmail(
+        *,
+        sender: str,
+        dests: list[str],
+        cc: list[str],
+        bcc: list[str],
+        subject: str,
+        body: str,
+        headers: dict[str, str],
+        attachments: list[dict[str, bytes]],
```
Comment (sveneberth) [inline src/viur/core/email.py:607]:
```suggestion
        attachments: list[Attachment],
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

**Review by sveneberth — APPROVED** [review]

**Pull request merged**